utils/queuer.go

```diff
@@ -67,13 +79,19 @@ func (q GeneralKVQueue) Dequeue(value codec.ProtoMarshaler, filter ...func(value
 		return false
 	}
 
-	if len(filter) > 0 && !filter[0](value) {
-		return false
-	}
+	pass, stop := filter(value)
+	switch {
+	case pass:
+		q.store.Delete(KeyFromBz(iter.Key()))
```

Comment:
`pass` is a bit confusing for the name. It can mean to `go past`/ignore it. Maybe use `skip` instead?